api/src/main/java/com/netflix/iceberg/io/FileIO.java

```diff
@@ -47,4 +46,25 @@
    * Delete the file at the given path.
    */
   void deleteFile(String path);
+
+  /**
+   * Get an {@link OutputFile} to write bytes for a new table metadata file with the given name.
+   */
+  OutputFile newMetadataOutputFile(String fileName);
+
+  /**
+   * Get an {@link OutputFile} for writing bytes to a new data file for this table.
+   * <p>
+   * The partition values of the rows in this file may be used to derive the final location of
+   * the file.
+   */
+  OutputFile newDataOutputFile(
```

Comment:
Same here, I think `@param` docs will help clarity. Also adding an example to get the path from the spec and partition data would help.